src/vaadin-grid.html

```diff
@@ -549,6 +549,7 @@
         if (row.id !== 'outersizer' && row.id !== 'fixedsizer') {
           row.hidden = true;
         }
+        columns = columns.filter(column => !column.hidden);
```

Comment:
nit: Instead of reassigning the `column` argument, use
```
columns
  .filter(column => !column.hidden)
  .forEach(...)`

Reply:
Done.